packages/openapi-generator/src/parser/operation.ts

```diff
@@ -47,13 +71,29 @@ export function parseParameters(
 ): OpenApiParameter[] {
   // TODO: What if this is a reference? What does OpenApi do?
   // TODO: What about oneof and other operations?
-  return (
-    operation.parameters
-      ?.map(param => resolveObject(param, refs))
-      .map(param => ({
-        ...param,
-        type: getType(resolveObject(param.schema, refs)?.type?.toString())
-      })) || []
+  return filterDuplicates(
+    operation.parameters?.map(param => resolveObject(param, refs)) || []
+  ).map(param => ({
+    ...param,
+    type: getType(resolveObject(param.schema, refs)?.type?.toString())
+  }));
+}
+
+export function filterDuplicates(
+  parameters: OpenAPIV3.ParameterObject[]
+): OpenAPIV3.ParameterObject[] {
+  return parameters.reduce(
```

Comment:
why you do not use a `parameters.filter(param=>param.name === currentParam.name && param.in === currentParam.in) ?